packages/oauth-client/src/EarthdataLoginClient.ts

```diff
@@ -0,0 +1,180 @@
+import got, { HTTPError, Response } from 'got';
+
+import { OAuthClient } from './OAuthClient';
+import { EarthdataLoginError } from './EarthdataLoginError';
+
+type VerifyTokenResponse = Response<{uid: string}>;
+
+type EarthdataLoginErrorResponse = Response<{error: string}>;
+
+const isHttpForbiddenError = (error: unknown) =>
+  error instanceof HTTPError && error.response.statusCode === 403;
+
+const isHttpBadRequestError = (error: unknown) =>
+  error instanceof HTTPError && error.response.statusCode === 400;
+
+const httpErrorToEarthdataLoginError = (httpError: HTTPError) => {
+  const response = <EarthdataLoginErrorResponse>httpError.response;
+
+  switch (response.body.error) {
+    case 'invalid_token':
+      return new EarthdataLoginError('InvalidToken', 'Invalid token');
+    case 'token_expired':
+      return new EarthdataLoginError('TokenExpired', 'The token has expired');
+    default:
+      return new EarthdataLoginError(
+        'UnexpectedResponse',
+        `Unexpected response: ${httpError.response.body}`
+      );
+  }
+};
+
+const validateUrl = (urlString: string) => {
+  // eslint-disable-next-line no-new
+  new URL(urlString);
+};
+
+/**
+ * A client for the Earthdata Login API
+ */
+export class EarthdataLoginClient extends OAuthClient {
+  readonly clientId: string;
+  readonly clientPassword: string;
+  readonly earthdataLoginUrl: string;
+  readonly redirectUri: string;
+
+  /**
+   * @param {Object} params
+   * @param {string} params.clientId - see example
+   * @param {string} params.clientPassword - see example
+   * @param {string} params.earthdataLoginUrl - see example
+   * @param {string} params.redirectUri - see example
+   *
+   * @example
+   *
+   * const oAuth2Provider = new EarthdataLoginClient({
+   *   clientId: 'my-client-id',
+   *   clientPassword: 'my-client-password',
+   *   earthdataLoginUrl: 'https://earthdata.login.nasa.gov',
+   *   redirectUri: 'http://my-api.com'
+   * });
+   */
+
+  constructor(
```

Comment:
Same comments as for Cognito client.
1. Use `loginUrl` instead of `earthdataLoginUrl`
2. Use object deconstruction to get the parameters.

Reply:
So I was debating this change. It would be cleaner to use `loginUrl` but I was reluctant to change the interface for `EarthdataLoginClient`, which uses `earthdataLoginUrl`. I suppose it isn't that risky if the change is only limited to our code here in the core repo. I can update all instances of the `EarthdataLoginClient`. Is there a possibility that `EarthdataLoginClient` is used outside of our code though? It's in a public package so change from `earthdataLoginUrl` to `loginUrl` would be a breaking change. 

Thoughts? I'm not opposed, just a little concerned about the implications.